cli/azd/pkg/account/manager.go

```diff
@@ -0,0 +1,250 @@
+package account
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/azcli"
+	"golang.org/x/exp/slices"
+)
+
+// Manages azd account configuration
+// `az cli` is not required and will only be called `azd` default have not already been set.
+type Manager struct {
+	azCli azcli.AzCli
```

Comment:
This introduces direct coupling to `azcli` which itself is a large interface, and we'll have to end up redesigning this part when we actually remove `azcli`.

I wonder if we could instead:
1. `Manager` should take in a list of `AccountProvider`
2. We would create `Manager` with `{azdAccountProvider, azCliAccountProvider}` if `az cli` is found. Otherwise, it would be `{azdAccountProvider}`.

Reply:
I'm also perfectly fine with deferring it until we need to, just calling it out that a refactoring will need to happen.

Reply:
Agree that the `AzCli` interface is large and also agree with breaking it up but I would prefer we defer it from this PR.